### PR TITLE
Converted to typescript

### PR DESCRIPTION
**Pull request by gforge**
Thanks for this awesome package. I previously did the same thing manually but without your elegant solution. I'm currently working in TypeScript and I noticed that your project didn't have any definitions. Instead of writing some hard to maintain .d.ts files I figured that I just rewrite the library in TS and hope that you find it to your liking :grinning: 

The basic changes are:

- Changed package.json script to use babel together with typescript
- Renamed all js to ts and added proper types (the only code that I had to touch was for `GraphQLLocalStrategy` where I figured that options should not be reset to {} - just using an else makes also the code intention clearer).
- I changed linting to airbnb typescript eslint
- There is also a types file for common types where the PassportRequest is probably the most important

All the tests and linting pass

### CODE REVIEW

**Comment by gforge** [comment]
Hey again, fixed some minor things. Seems to work with my other repo now.

**Comment by jkettmann** [comment]
Thanks a lot for your PR. I'm currently a bit busy. But I'll have a look in the next days for sure

**Comment by gforge** [comment]
Great, no stress. It's a big PR and annoyingly then js -> ts makes it harder to follow the changes line-by-line.

**Comment by gforge** [comment]
Hey, how's the review coming along?

**Comment by nmklotas** [comment]
Would be nice to get this merged ;)

**Comment by jkettmann** [comment]
@gforge @nmklotas Thanks a lot for your patience. Had a very busy time. I'll try to merge this until the end of the year.

File: src/buildContext.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  req: express.Request;
+  res?: express.Response;
+}
+
+const buildCommonContext = (req: express.Request, additionalContext: {}): Context => ({
+  isAuthenticated: () => req.isAuthenticated(),
+  isUnauthenticated: () => req.isUnauthenticated(),
+  getUser: () => req.user,
+  authenticate: (strategyName: string) => {
+    throw new Error(`Authenticate (${strategyName}) not implemented for subscriptions`);
```
Comment (jkettmann) [inline src/buildContext.ts]: Good idea!

Comment (gforge) [reply src/buildContext.ts]: Thanks! I've been working with the package with unit tests and there are some minor things to the Readme that I think may help (I probably spent an hour before I got `apollo-server-testing` to work). Also, the `PassportRequest` should be a generic accepting a the User interface that is returned.

File: README.md [hunk 2]
```diff
@@ -151,3 +159,32 @@ const resolvers = {
...
+}
+```
+
+In your GraphQL functions you can the reference the passport context using:
+
+```ts
+const something = async (
+  parent: unknown,
+  data: SomeData,
+  context: PassportRequest
```
Comment (jkettmann) [inline README.md]: I guess I was too quick. This is not correct, the context shouldn't be of type `PassportRequest`. I guess we need an additional type or export the `Context` type from `buildContext` as well.

Can you create a new PR for this? @gforge

**Comment by gforge** [comment]
Sure, no worries.

**Comment by crutch12** [comment]
@gforge Thx for ts support.
But there is a probles - is run with ts-node app doesn't start bacause of unexisted main resolver:
```
"main": "lib/index.js"
```
in package.json

**Pull request merged**